### PR TITLE
[Data2Vec Text] Correct Data2Vec Text

### PR DESCRIPTION
**Pull request by patrickvonplaten**
# What does this PR do?

Data2Vec Text has no `ForMaskedLM` model. The point of the paper is to not have this pretraining objective, but the Data2Vec pretraining method as shown here: https://huggingface.co/facebook/data2vec-text-base . 

This pretraining method is too complicated though to add it right away. So for now we'll just add the fine-tuned checkpoints.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Review by sgugger — COMMENTED** [review]
We can't just remove a class that is in the main init and was in the release, and I also don't see why we should remove it. GPT-2 is not meant for token classification if you read its paper, yet we still have the ` GPT2ForTokenClassification`.

Likewise, even if this model was not pretrained using masked language modeling, I don't see why we should delete the architecture.

**Comment by patrickvonplaten** [comment]
> We can't just remove a class that is in the main init and was in the release, and I also don't see why we should remove it. GPT-2 is not meant for token classification if you read its paper, yet we still have the ` GPT2ForTokenClassification`.
> 
> Likewise, even if this model was not pretrained using masked language modeling, I don't see why we should delete the architecture.

Ah yeah this one was already in the release :-/ 
We haven't promoted it yet, so not sure anybody really uses the models yet, but yeah not great if the class already existed in a release. 

It's a borderline error to me as `ForMaskedModeling` is rarely used for fine-tuning IMO, but more or less solely for pretraining. We might in the future add the correct `ForPreTraining` class, so this could be a bit confusing, but overall probably not worth the hussle here so fine with just leaving it.

cc @edugp @mrm8488
